drift.py

```diff
@@ -1,146 +1,115 @@
+import json
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from datetime import datetime
+from subprocess import run as subprocess_run
+
+import requests
 from helicase import Helicase
-from datetime import *
-import argparse
-import json, requests
-import subprocess, sys
+from spack.cmd.diff import compare_specs
+from spack.spec import Spec
+from spack.version import ver
 
 # Define SUCCESS for comparing command return codes.
 SUCCESS = 0
-args = None
 
+@dataclass
 class Result:
-    abstract:str
+    name:str
+    version:str
     commit:str
     tags:list
     timestamp:str
-    spec:str
-    version:str
-
-    def __init__(self):
-        self.abstract = ""
-        self.commit = ""
-        self.tags = []
-        self.timestamp = ""
-        self.spec = ""
-        self.version = ""
-
 
 class DriftAnalysis(Helicase):
-    specs = []
-    last = {}
-    def __init__(self, specs):
-        self.specs = specs
-        for spec in specs:
-            self.last[spec] = ""
+    def __init__(self, specs=None):
+        """Setup the last cache and specs used for the analysis."""
+
+        self.last = {}
+        self.specs = specs or []
```

Comment:
nitpick: if the user passes an empty list and intends for you to use *that specific one*, you'll end up overriding the user with your own.  The way to avoid that is:

```python
self.specs = [] if specs is None else specs
```

See the difference?  I do not think this needs changing, but you should know in case a bug pops up that needs it. IMO it's ugly and the user shouldn't rely on this and should just refer to the `specs` member of `DriftAnalysis` if they need the actual list.